sdk/src/main/java/com/hedera/hashgraph/sdk/AccountId.java

```diff
@@ -37,6 +44,33 @@ public AccountId(@Nonnegative long shard, @Nonnegative long realm, @Nonnegative
         this.shard = shard;
         this.realm = realm;
         this.num = num;
+        this.network = null;
+        this.checksum = null;
+    }
+
+    AccountId(@Nonnegative long shard, @Nonnegative long realm, @Nonnegative long num, @Nullable NetworkName network, @Nullable String checksum) {
+        this.shard = shard;
+        this.realm = realm;
+        this.num = num;
+        this.network = network;
+
+        if (network != null) {
+            if (checksum == null) {
+                this.checksum = EntityIdHelper.checksum(network.toString(), shard + "." + realm + "." + num);
+            } else {
+                this.checksum = checksum;
+            }
+        } else {
+            this.checksum = null;
+        }
+    }
+
+    public static AccountId withNetwork(@Nonnegative long num, NetworkName network) {
```

Comment:
I think we need to discuss this more. `AccountId.withNetwork(15, NetworkName.MAINNET)` doesn't feel right to me.

---

The goal here is to enable printing an account ID with a checksum - which requires knowing the network. Users will have an `AccountId` type already (most likely) and will want to print the ID with a checksum against the current network. The current network in this case most likely means a network from a `Client` instance. 

 - `client.formatEntityId(accountId)`
 - `accountId.withChecksum(client).toString()`

cc @lbaird @steven-sheehy @Sean-Tedrow-LB @danielakhterov @andrix10 

Pinging some from Hedera who might be interested in how this API gets figured out. As printing a valid ID with a  checksum is almost as important as validating it with the checksum.

Reply:
What we can do is set the network on the entity ID implicitly when executing with a client that has a network set. So if someone constructs a client using `Client.for[Mainnet|Testnet|Previewnet]()`, but uses entity IDs without a network, we can set the network accordingly when they execute any request.